flaskr/api/views.py

```diff
@@ -0,0 +1,39 @@
+from flask import abort, jsonify, current_app
+import json
+from . import app
+
+
+"""
+Return json for all courses under <spec> specialisation
+in <sem> semester (X1|S1|S2)
+"""
+@app.route('/specialisation/<spec>/<sem>')
+def specialisation(spec, sem):
+    sem = sem.upper()  # accept lower case request
+
+    db_file = app.root_path + '/specialisation/db.json'
+    # Try to read the json from the db file, fail gracefully and log an error
+    try:
+        db = json.load(open(db_file, 'r'))
```

Comment:
Can this decoding be extracted to its own function (or module)? It would make this more maintainable into the future.

Reply:
lol cbs